src/remarx/sentence/corpus/base_input.py

```diff
@@ -60,6 +60,16 @@ def get_text(self) -> Generator[dict[str, str]]:
         """
         raise NotImplementedError
 
+    def get_line_number(
+        self, chunk_info: dict[str, Any], char_idx: int, sentence: str
+    ) -> int | None:
+        """
+        Hook method for subclasses to override to provide line number for a sentence.
+
+        :returns: Line number (1-indexed) if available, None otherwise
+        """
+        return None
+
```

Comment:
Let's rename `get_extra_metadata` or similar. Method should return a `dict` of fields and values to include; default implementation returns `None` or empty dict, whichever seems easier.